src/main/java/com/spring/app/domain/Country.java

```diff
@@ -66,7 +66,45 @@
                                 "                              AND (vacancy.date_closed ISNULL OR vacancy.date_closed >= :to_date))" +
                                 "    WHERE country.name = :country_name " +
                                 "    GROUP BY company.name) " +
-                                "AS count_table"
+                                "AS count_table"),
+
+                @NamedNativeQuery(
+                        name = "EMPLOYEE_PROFILES",
+                        query = "SELECT " +
+                                "       country_name_field       AS country, " +
+                                "       AVG(avgSalary)           AS salary," +
+                                "       AVG(avgAge)              AS age, " +
+                                "       AVG(avgExp)              AS experience, " +
+                                "       AVG(number_of_employees) AS number_of_employees " +
+                                "FROM (SELECT COUNT(country.id)                                                                          AS number_of_employees, " +
+                                "             country.name                                                                               AS country_name_field, " +
+                                "             AVG(vacancy.salary)                                                                        AS avgSalary, " +
+                                "             AVG((cast(extract(year FROM cast(:to_date AS DATE)) AS int)) - employee.birth_year)        AS avgAge, " +
```

Comment:
Good, but let's make more performance-optimise. You can extract year at the top select statement. This will be calculated only once, when the inner select completes.

Reply:
i did a new one!

Reply:
you will like it